pkg/config/piped.go

```diff
@@ -256,6 +261,58 @@ type HelmChartRepository struct {
 	Password string `json:"password"`
 	// Whether to skip TLS certificate checks for the repository or not.
 	Insecure bool `json:"insecure"`
+
+	// Remote address of the Git repository used to clone Helm charts.
+	// e.g. git@github.com:org/repo.git
+	GitRemote string
+	// The path to the private ssh key file used while cloning Helm charts from above Git repository.
+	SSHKeyFile string `json:"sshKeyFile"`
+}
+
+func (r *HelmChartRepository) IsHTTPRepository() bool {
+	return r.Name != "" && r.Address != ""
+}
+
+func (r *HelmChartRepository) IsGitRepository() bool {
+	return r.GitRemote != ""
+}
+
+func (r *HelmChartRepository) Validate() error {
+	if r.IsHTTPRepository() {
+		if r.Name == "" {
+			return errors.New("name must be set")
+		}
+		if r.Address == "" {
+			return errors.New("address must be set")
+		}
+	}
+
+	if r.IsGitRepository() {
+		if r.GitRemote == "" {
+			return errors.New("gitRemote must be set")
+		}
+	}
+	return nil
+}
```

Comment:
This part blows my mind. I just feel like in case of nothing is provided (no Name, Address, GitRemote), this validation will return nil, which means the config is valid, but it looks like we need at least one of those fields to be configured, right?

Reply:
Ah right. It should be errored in that case. Let me fix.

Reply:
Addressed it!